### ISSUE TITLE
[Remote Debug] Debug C# and PY in parallel

### ISSUE THREAD

**Issue by vsfeedback** [microsoft/PTVS#6907]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Remote-Debug-Debug-C-and-PY-in-parall/1678063)._

---
[severity:It's more difficult to complete my work]
We wanted to use same IDE session to debug 2 processes remotely: .NET 5 and python. 
Both process managed to be connected as depicted below: 
! [image.png] (https://aka.ms/dc/image?name=B051860eda140451093ce8aefd6ba7d56637816066431318195_20220228-085722-image.png&tid=051860eda140451093ce8aefd6ba7d56637816066431318195)

Managed to add breakpoints into C# codes. 
However, unable to add breakpoints into PY codes with error below.
! [image.png] (https://aka.ms/dc/image?name=Bf770d28d0cc645f78cc301df9dd7f6a1637816069350350312_20220228-090213-image.png&tid=f770d28d0cc645f78cc301df9dd7f6a1637816069350350312)

If we detach from .NET 5 process, then we will be able to add hit PY codes correctly. 
! [image.png] (https://aka.ms/dc/image?name=B37a88bc9bb154c76b221fd835c959f1f637816070460457925_20220228-090404-image.png&tid=37a88bc9bb154c76b221fd835c959f1f637816070460457925)

Issue is only observable when done on remote process. It is working fine when processes are local.

---
#### Feedback Bot on 2/28/2022, 02:06 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by int19h** [comment]
Can you clarify how exactly remoting was set up? Looking at the screenshots, it appears that you might be using VS Remote Debugging Monitor? (msvsmon.exe)

**Comment by gaikhi** [comment]
Yes, we are using msvsmon at the target system.

**Comment by int19h** [comment]
It appears that when we switched to the new DAP-based Python debugger (debugpy), we broke the msvsmon scenario. Technically it might still be possible to set it up by carefully laying out the right assemblies, but it's not something that we consider intentionally supported at this point.

Try using [debugpy remoting](https://docs.microsoft.com/en-us/visualstudio/python/debugging-python-code-on-remote-linux-machines?view=vs-2022) for your Python process instead.
